### DISCUSSION TITLE
Generate output in documentation format

### DISCUSSION THREAD

**Discussion by bryanenders** [Quick/Quick#1077] [Ideas]
RSpec allows me to generate documentation using `--format doc` or `-fd`. As a developer on a team, I want to be able to generate this sort of documentation using Quick:

```
A dolphin
  click
    when it’s not near anything interesting
      emits only one click ✅
    when it’s near something interesting
      emits a series of clicks ❌
```

**Comment by ashfurrow** [comment]
At Artsy we use [xcpretty](https://github.com/supermarin/xcpretty), which works pretty well. Maybe it can do what you're looking for?

**Comment by nikita-leonov** [comment]
Xcpretty will generate a better test reports, but it still will be based on XCTest output which is  
`✓ -[BuddiesRequestsSpec Buddies_Requests__for_a_first_newly_registered_users__Hopper_s_Empty_Buddies_Requests_List__is_empty_] (2289 ms)`
As far as I understood ExampleGroup / Example runs does not have a printout of descriptions, as it will be kind of more readable duplication with XCTest output. Meanwhile it could be nice feature to have a custom formatting of output for a further use in custom reporting tools like XCPretty to achieve results as @endersstocker mentioned. I had similar formatting request in my list as well, but didn't approach it yet as it is not a top priority one.

**Comment by modocache** [comment]
Yes, this is definitely something we want to support as part of #9.

**Comment by modocache** [comment]
I actually wrote a short novel on the subject here: https://github.com/kiwi-bdd/Kiwi/issues/489#issuecomment-38953417
